packages/connect/src/core/index.ts

```diff
@@ -1219,9 +1218,47 @@ const disableWebUSBTransport = async () => {
     }
 };
 
+const initCore = async (
+    settings: ConnectSettings,
+    onCoreEvent: (message: CoreEventMessage) => void,
+    logWriterFactory?: () => LogWriter,
+) => {
+    const core = new Core();
+    let promise: Promise<void>;
+
+    // do not send any event until Core is fully loaded
+    // DeviceList emits TRANSPORT and DEVICE events if pendingTransportEvent is set
+    const eventThrottle = (...args: Parameters<typeof onCoreEvent>) =>
+        promise
+            .then(() => {
+                setTimeout(() => onCoreEvent(...args), 0);
+            })
+            .catch(() => {});
+
+    promise = core.init(settings, eventThrottle, logWriterFactory);
+    await promise;
+
+    // Core initialized successfully, disable throttle
+    core.on(CORE_EVENT, onCoreEvent);
+    core.off(CORE_EVENT, eventThrottle);
+
+    return core;
+};
+
+const disposeCore = (core: Core) => {
+    core.dispose();
+};
+
 /**
  * State initialization
  */
 export const initCoreState = () => {
-    return initCoreManager(new Core());
+    const { get, getPending, getOrInit, dispose } = createLazy(initCore, disposeCore);
+
+    return {
+        getCore: get,
+        getInitPromise: getPending,
+        getOrInitCore: getOrInit,
```

Comment:
nitpick/discussion i'd probably wouldn't do the renaming here to increase homogenity of codebase wherever this is consumed